src/commands/deploy.ts

```diff
@@ -54,41 +60,116 @@ $ bump deploy FILE --dry-run --doc <doc_slug> --token <your_doc_token>
   async run(): Promise<void> {
     const { args, flags } = this.parse(Deploy);
 
-    const [dryRun, documentation, token, hub, autoCreate, documentationName, branch] = [
+    const [
+      dryRun,
+      documentation,
+      token,
+      hub,
+      autoCreate,
+      interactive,
+      documentationName,
+      branch,
+    ] = [
       flags['dry-run'],
-      /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
-      flags.doc!,
+      flags.doc,
       /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
       flags.token!,
       flags.hub,
       flags['auto-create'],
+      flags['interactive'],
       flags['doc-name'],
       flags.branch,
     ];
     const file = await statSync(args.FILE);
 
-    if (file.isDirectory()) {
-      throw new Error(
-        'Deploy with a directory is not yet implemented. Please wait while we are working on it!',
-      );
-    } else {
-      await this.deploySingleFile(
+    if (file.isDirectory() && hub) {
+      await this.deployDirectory(
         args.FILE,
         dryRun,
-        documentation,
         token,
         hub,
         autoCreate,
+        interactive,
         documentationName,
         branch,
       );
+    } else {
+      if (documentation) {
+        const api = await API.load(args.FILE);
+        this.d(`${args.FILE} looks like an ${api.specName} spec version ${api.version}`);
+
+        await this.deploySingleFile(
+          api,
+          dryRun,
+          documentation,
+          token,
+          hub,
+          autoCreate,
+          documentationName,
+          branch,
+        );
+      } else {
+        throw new RequiredFlagError({ flag: Deploy.flags.doc, parse: {} });
+      }
+    }
+
+    return;
+  }
+
+  private async deployDirectory(
+    dir: string,
+    dryRun: boolean,
+    token: string,
+    hub: string,
+    autoCreate: boolean,
+    interactive: boolean,
+    documentationName: string | undefined,
+    branch: string | undefined,
+  ): Promise<void> {
+    let documentations: DocumentationConfig[] = [];
+    const service = new Directory(dir);
```

Comment:
Could the naming "service" could be confusing later ?

Reply:
I've refactored a bit an changed the class to `ContractDirectory`.

I've also changed the logic around the class to keep the data context **inside** the class (previously the `documentations` variable in the deploy command, now a `ContractDirectory` instance variable named `this.contracts`).

With that done, I believe we can keep all methods inside the `ContractDirectory` class (without the need to extract some methods prematurely that are not “specific to a contract directory”.

If you still think some methods need to be extracted in some helper classes, please let me know. For now I'll resolve all other discussions about this subject. :pray:

Reply:
> With that done, I believe we can keep all methods inside the `ContractDirectory` class (without the need to extract some methods prematurely that are not “specific to a contract directory”.
> 
> If you still think some methods need to be extracted in some helper classes, please let me know. For now I'll resolve all other discussions about this subject. pray

I've changed my mind and extracted a `utils/file.ts` class with the “file naming” concern, it was a good idea, thanks :+1:!